api/v1/db/oracledb/player-dao.js

```diff
@@ -83,18 +83,63 @@ const cleanPlayerCardsByPlayerId = async (playerId, connection) => {
   return result;
 };
 
-const deletePlayerByPlayerId = async (playerId) => {
-  const connection = await conn.getConnection();
+const deletePlayerByPlayerId = async (playerId, passedConnection) => {
+  const isPassedConnection = Boolean(passedConnection);
+  const connection = isPassedConnection ? passedConnection : await conn.getConnection();
   try {
     await cleanPlayerCardsByPlayerId(playerId, connection);
     const sqlParams = [playerId];
     const deleteSqlQuery = `
     DELETE FROM PLAYERS WHERE PLAYER_ID = :playerId
     `;
-    const result = await connection.execute(deleteSqlQuery, sqlParams, { autoCommit: true });
-        return result;
+    const result = await connection.execute(deleteSqlQuery, sqlParams,
+      { autoCommit: !isPassedConnection });
+    return result;
   } finally {
-    connection.close();
+    if (!isPassedConnection) {
+      connection.close();
+    }
+  }
+};
+
+
+const deletePlayersByGameId = async (gameId, connection) => {
+  const sqlParams = [gameId];
+  const playerSqlQuery = `
+  SELECT PLAYER_ID FROM PLAYERS P
+  WHERE GAME_ID = :gameId
+  `;
+  const rawPlayersResponse = await connection.execute(playerSqlQuery, sqlParams);
+  const playerIds = _.map(rawPlayersResponse.rows, player => (player.PLAYER_ID));
+  const bindString = playerIds.map((name, index) => `:${index}`).join(', ');
+  if (!_.isEmpty(playerIds)) {
+    const deletePlayerCardSqlQuery = `
+    DELETE FROM PLAYER_CARDS WHERE PLAYER_ID IN (${bindString})`;
+    await connection.execute(deletePlayerCardSqlQuery, playerIds);
+
+    const deletePlayersSqlQuery = `
+    DELETE FROM PLAYERS WHERE PLAYER_ID IN (${bindString})`;
```

Comment:
You should be able to nest the `SELECT` statement above in these two `DELETE` statements to avoid having to make an extra DB query.